store/src/java/com/zimbra/cs/mailbox/ContactAutoComplete.java

```diff
@@ -756,6 +748,49 @@ public void addMatchedContacts(String query, Map<String, ? extends Object> attrs
         }
     }
 
+    /**
+     * Identifies the eligible results for GAL Autocomplete for an account
+     * @param attrs
+     * @return
+     * @throws ServiceException
+     */
+    protected List<String> extractEligibleEmailsListForAccount(Map<String, ?> attrs) throws ServiceException {
+        List<String> eligibleEmailsForAccount = new ArrayList<>();
+        String primaryEmail = getFieldAsString(attrs, ContactConstants.A_email);
+        Account account = Provisioning.getInstance().get(Key.AccountBy.name, primaryEmail);
+        populateEmailsList(attrs, eligibleEmailsForAccount, account);
+        return eligibleEmailsForAccount;
+    }
+
+    /**
+     * populats the eligible
+     * @param attrs
+     * @param eligibleEmailsForAccount
+     * @param account
+     * @throws ServiceException
+     */
+    private void populateEmailsList(Map<String, ?> attrs, List<String> eligibleEmailsForAccount, Account account)
```

Comment:
`populateEligibleEmails()` in place of `populateEmailsList()`?

Reply:
fixed